### ISSUE TITLE
Lab 8: Tree Fractals

### ISSUE THREAD

**Issue by lmock** [cs10/bjc-r#430]
Please comment below to leave us your lab reviews! Remember, please follow this format:

```
Your Name:

Overview of Lab Comments: i.e. Was this lab good or bad? Too fast? Too Slow? Does it need something new?

For specific issues: (you should copy this section as many times as necessary)
Page Link:
Type of Issue: i.e. (code) bug, grammar, typo, lack of clarity, etc
Severity: i.e. "prevents students from continuing", "minor inconvenience", etc
Comments: Description of the problem, what can be done to fix it, etc.
```

**Comment by ArielleSpencer** [comment]
Your Name: Arielle Spencer

Overview of Lab Comments: Great overall, god speed and content until the c-curve which seemed much more difficult and unguided than the rest of the lab

For specific issues: 
Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/fractals/c-curve.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=9
(C- Curve)
Type of Issue: lack of clarity?
Severity: difficult
Comments: Was really difficult to do, I think there should be a few more hints and step by step directions as it was mostly just based off a few pictures. Also, maybe clarify the "pointing right" thing as in the pictures sometimes it was hard to tell, maybe could be "pointing in the same direction" or something like that

**Comment by rxia752** [comment]
Rowena Xia

Overview of Lab: This lab is very thorough in creating a variety of information and practice for the student.  I like that there was an abundance of quiz questions, because this tree recursion lab can be very difficult.  I really like this page http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/base-cases.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=7 because it provides a thorough point-by-point explanation for how this recursion works.  This is very helpful, especially for helping to make sure students understand this difficult and important concept.  I do think that there needs to be more guidance with some of the problems, such as fixing the crazy-tree as students may feel lost as to where to start.  Additionally, from my experience, this lab may be on the long side because tree recursion can be tricky and students can get stuck often, so it may be a bit of a struggle to finish what is given in the lab.  It would be good to either trim the lab down a little or just give some more guidance so it's easier for students (such as hints to make students think in the right direction).  

Specific Issues:
Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/fractals/c-curve.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=9
Type of Issue: lack of clarity
Severity: Major
Comments: There definitely needs to be more guidance here for the C-curve.  Students could be easily intimidated by this problem.  There needs to be more guidance in the forms of step by step instruction so that the student is not overwhelmed.

**Comment by 333lisajin** [comment]
Lisa Jin

Overview of Lab Comments: It is good. The content is really interesting and the pace is OK too.
For specific issues: (you should copy this section as many times as necessary)
Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/recursive-tree-part-2.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=1

Type of Issue: i.e. Too wordy
Severity: prevents students from continuing
Comments: Make it into 2 pages

**Comment by Hungukaroo** [comment]
Elizabeth Steger

I think this lab is really well done, it uses a good balance between telling the student what to do and throwing them in the deep end.
- [x] Not any real problems that I can see,
  Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/fractals/search-for-fractals-in-nature.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=10

Issue: This seems kind of random, when I finished the lab it was kind of like, "Here, look at some fractals." There's nothing really wrong with this but I would have appreciated a congratulations and maybe a segue like, "Now that you've seen what kind of beautiful fractals you can make with a computer, check out these amazing ones found in nature!" or something to that effect, I feel like it would make it a better conclusion.
Severity: extremely minor

**Comment by amrutayel** [comment]
Your Name: Amruta Yelamanchili

Overview of Lab Comments: This is one of the hardest labs, but also one of the most important. The page explaining base case should probably go along with “Recursive tree part 3.” It asks the students to identify the base case, but does not go into detail what this means. It will most likely take more than two hours for a majority of students to finish this lab.

For specific issues: 

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/treepart3.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=2
Type of Issue: too long of a problem
Severity: students won’t continue
Comments: I noticed last semester that a lot of students would get stuck on making a more realistic tree. They would focus on this for a long time, and they wouldn’t get to the more important parts of the lab. They also have not yet learned about changing angles and such in tree fractals, so they might get hung up on the fact that the “realistic tree” shown does have angles that aren’t even. Maybe ask them to make a more realistic tree after the “vary your tree” page. 

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/selftest1.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=3
Type of Issue: lack of clarity 
Severity: students might not understand
Comments: The previous tree example had a base case, but none of the questions asked here have code that has a base case, which might be confusing.

**Comment by msilva926** [comment]
Your Name: Melanie Silva

Overview of Lab Comments: This was a challenging lab which might take longer than 2 hours for a student to complete

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/fractals/snowflake.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=8
Type of Issue: Long Lab
Severity: minor inconvenience
Comments: The Snowflake followed by the C-Curve problem are challenging problems and students who just learned Recursion might not be able to complete the entire lab in 2 hours.

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/treepart3.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=2
Type of Issue: Lack of Clarity
Severity: i.e. minor inconvenience
Comments: I suggest that an explanation about what levels and size are and do would be helpful. Might be unclear to the student.

**Comment by maknegana** [comment]
Your Name: Michelle Han

Overview of Lab Comments: The lab was thorough. I felt that the lab was a bit too long for an introduction to recursion. I liked how there were a lot of quizzes-- it made concepts clearer. 

For specific issues: 
Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/treepart3.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=2
Type of Issue: Lack of clarity
Severity: Minor Inconvenience
Comments: Doesn't explain the function of a base case. Perhaps have this page before: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/base-cases.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=7

For specific issues: 
Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/fractals/c-curve.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=9
Type of Issue: Lack of Clarity
Severity: Getting stuck
Comments: Perhaps give more guidance throughout instead of leaving the students hanging. The beginning instructions were vague as well, "start facing right, end facing right."

**Comment by william0cheung** [comment]
William Cheung

Overview: I love this lab, it's probably one of the coolest ones out there. Tree fractals are awesome and it is presented in a very easy to understand way. Recursion is explained very nicely, especially with good visuals of the tree fractals. Also the quiz questions were very helpful.

Specific issues
Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/fractals/c-curve.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=9
Type of Issue: i.e. lack of clarity
Severity: minor inconvenience
Comments: Students might need to look at the code again for the snowflake on this page. It was presented last page but they might want to refer to it again.
Also maybe a wrap up of the base case and recursive case won't hurt at the end to sum up the point of fractals. Otherwise, this is one of my favorite labs!

**Comment by melvinhernandez** [comment]
Your Name: Melvin Hernandez

Overview of Lab Comments: I felt that the lab was really long. I thought that creating 'crazy-tree' and 'realistic tree' would have been a little difficult, especially because there were no hints. The lab is a fun way of introducing people to recursion.

For specific issues: 
Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/random-tree.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=6
Type of Issue: Difficult problem without any hints. 
Severity: Minor Inconvenience
Comments: It's a great puzzle that students can try to figure out, however, they may get frustrated if the lab does not offer them any guidance as to how they may try to approach the problem. Maybe try a hidden hint written in a white font for those who really need it.

For specific issues: 
Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/fractals/c-curve.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=9
Type of Issue: Lack of Clarity
Severity: Minor Inconvenience.
Comments: Give an insight on how the students can change the colors the way they are displayed in the pictures. Provide hints, perhaps?

**Comment by joelterry** [comment]
Your Name: Joel Terry

Overview of Lab Comments: A very effective, yet gradual introduction to recursion. The recursive tree exercise seemed like a lot of busy work at first, but then I remembered how strange the concept of recursion is to those who are first learning it. It's definitely a topic needs to be introduced carefully, and the tree exercise does a good job of that.

Page Links: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/fractals/snowflake.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=8
and
https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/fractals/snowflake.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=8https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/fractals/snowflake.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=8
Type of Issue: difficulty due to lack of clarity, possibly
Severity: 
Comments: The last two problems are extremely intimidating to someone who's new to recursion, unless you make it VERY clear that the first few levels (maybe even just the first two) are what you need to focus on when making these functions. This was stated explicitly in C-curve (maybe not stressed enough though), and not at all in snowflake.

**Comment by ghost** [comment]
Your Name: Anjli Shah

Overview of Lab: I think the lab definitely does a wonderful job of helping students understand tree fractals. However I think there could have been more guidance in terms of hints or thinking points to get them started on more difficult problems. It also might be helpful to put the breakdown of recursive code earlier. Overall, it was a well done lab.

For specific issues: 

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/recursive-tree-part-2.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=1 
Type of Issue: Clarity in Understanding
Severity: Minor Inconvenience
Comments: Since recursion is a pretty tricky concept it might be helpful to put some thinking points. For example, why does the recursive tree block need to know exactly how many levels to make? Why do we need to end the block at the starting position?

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/base-cases.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=7 
Type of Issue: Difficult in starting problem 
Severity: Can’t go on
Comments: It’s very difficult to figure out why the crazy tree block isn’t working. A hint/thinking questions might help.

**Comment by chrisk867** [comment]
Name: Chris Kilian

Overview of the lab: I liked the way that this lab is set up. Fractals and recursion are very hard topics to wrap your head around when you first start them, so expecting the students to be able to do fractals right off the bat would be wrong. The slowly walks the student how to do the tree fractal and even gives sort of a thought process on how we reached the conclusion. I liked that this lab was slow and gave the answers at the beginning because it is a good way for the student to learn how to think about these problems.
- [x] Page Link:http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/fractals/search-for-fractals-in-nature.html?topic=berkeley_bjc/recur/recursion-trees-fractals.topic&step=10&noassignment=true&noreading=true&novideo=true&course=cs10_fa14.html
  Type of Issue: coherence
  Severity: minor
  Comments: This seems like a really random slide just lopped onto the end of the lab. I don't think it needs to go, but it just seems really random.

**Comment by numfah23** [comment]
Your Name: Natnicha Vanitchanant

Overview of Lab Comments: Overall, the lab was good and the topic of recursion was presented really well. It took a while to copy down all the blocks because the link to snap wasn’t provided, but I think it would be better for the students to get to physically build the blocks themselves. Something that bothered me personally was that the first tree blocks were purple and all the other blocks were blue (they should all be blue because they are related to motion?). Also, the order of the size and level variables weren’t consistent (ex. tree had level before size, crazy-tree and snowflake had size before level).

For specific issues:
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/recursive-tree-part-1.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=0
  Type of Issue: Lack of clarity
  Severity: Minor inconvenience
  Comments: At the end of the page, it says “At this rate you’ll never be able to make the beautiful version of the tree in this century”. It would be nice to clarify why this would be a problem by talking about how many times you would need to repeat the move and turn blocks.
- [x] Page Link:http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/recursive-tree-part-2.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=1
  Type of Issue: i.e. Typo
  Severity: Minor inconvenience
  Comments: “So it makes sense to wonder if we CAN’T replace them all with a single tree block”. CAN’T should be replaced with CAN.
- [ ] Page Link:http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/base-cases.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=7
  Type of Issue: i.e. Lack of clarity
  Severity: Minor inconvenience
  Comments: : “Lines 2-4 are the Base Case, in which we handle a situation so simple that we don’t need to call the block recursively”. Instead of saying “a situation so simple”, it would be more clear to say “the most simple situation(s)” since this is technically what a base case is.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/fractals/snowflake.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=8
  Type of Issue: i.e. Lack of clarity
  Severity: Minor inconvenience
  Comments: : For the images of the call to the snowflake block with different level inputs, it would be nice to have the levels written next to the pictures.
- [ ] Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/fractals/c-curve.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=9
  Type of Issue: i.e. Grammar
  Severity: Minor inconvenience
  Comments: : “Below the base case is that the sprite draws a single line. The sprite starts facing right and faces right at the end” The first sentence is awkward. To fix it, change it to something like “In the image below, the base case is that the sprite draws a single line”. The second sentence sounds better like this “The sprite starts facing right and ends facing right”.

**Comment by strollinghome** [comment]
Carlos Flores.

The lab was well done. I think we should regress from making students generate a tree and focus more on the recursion part of it. I feel like students could get caught up trying to make the tree look exactly like the specified one and not think about the recursive process. Other than that the pace was good and overall it explains everything correctly.

The link at the end to fractals in nature is pretty cool.

**Comment by Frankgoji** [comment]
Franklin Lee

Overview of Lab Comments:
The recursion and fractals lab is filled with examples that I think will grab students' interest (at least more than finding an item in a list). The exploration of how the recursive call works, and how Snap manages the recursion, is executed slowly and carefully, which I think is a good thing. Finally, the lab also explores the basic structure of a recursive function, which is arguably the most important part of the lab, highlighting important aspects such as the base case and the recursive call so that students will recognize it. The length of the lab is appropriate, I think. It isn't too long, since it isn't jam-packed full of examples, but it proceeds slowly through the examples it does have, so that they're more useful.

For specific issues:

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/recursive-tree-part-1.html?topic=berkeley_bjc/recur/recursion-trees-fractals.topic&step=0&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html
Type of Issue: Unclear code, description
Severity: mild-moderate
Comments: The student's may not immediately recognize that having no input in the subtraction reporter is the same as putting a zero in the slot. It may be constructive and useful to insert that zero so that at first glance it makes sense mathematically. Also, the paragraph after that explains itself saying that the script "moves the sprite back to its original position." While we do mention that we must move the sprite back to the original position AND the original orientation later in the sentence, it might help reinforce the fact that the direction matters if it is in the same place and with the same emphasis (italics).

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/selftest1.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=3
Type of Issue: unclear code
Severity: mild
Comments: In the second and third questions, the call to the tree block has no value for levels, even though the third questions specifies that levels is set to -4. Since there is no level input in the second question, it may cause some to think that levels is set to 0, which would draw nothing, rather than thinking about what would happen if size is negative.

**Comment by erikdahlquist** [comment]
Erik Dahlquist

I think this lab is exceptional. After a somewhat dry/random patch of labs focusing on algorithms and robots, the Recursion lab is well-paced, interesting, and can be genuinely fun. Recursion is a fascinating piece of technological wizardry, and I think this lab does a superb job gently introducing students to it, making sure they understand the underlying concepts, and then adding in more difficult ideas. It has relatively few mishaps, all of which are relatively minor.
Seriously, I cannot praise the pacing of this lab enough. The literally step-by-step tree-fractal which shows how much easier recursion can be is great. As are the multiple examples of recursive fractals that constantly increase in difficulty and decrease in hand-holding.
Perhaps what could be fun to add to the lab are completely optional "challenge fractals" at the end.

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/recursive-tree-part-1.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=0
Type of Issue: Mouse pop-up?
Severity: Minimal. Won't be noticed by many.
Comments: When you mouse over certain images, namely the pretty tree fractal and the move block in step 1, text pops up by the mouse saying "there should be an image here." There is an image there. Is this a remnant that is no longer needed?

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/recursive-tree-part-1.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=0
Type of Issue: Old sprite
Severity: Minor
Comments: The sprite shown in steps 2 and 3 is an unfamiliar circle instead of the turtle sprite we know and love. Perhaps it should have the modern turtle sprite, especially because the turtle sprite shows direction, which is important when making fractals. 

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/selftest1.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=3
Type of Issue: Unclear question
Severity: Minor
Comments: For question 2, "levels" is blank. You may want to put a number in there so students don't get confused on what the real issue is.

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/random-tree.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=6
Type of Issue: Time-sucking
Severity: Moderate
Comments: Could we give them the buggy Random Tree code to work with? While maybe good practice, constructing it from scratch might not be the best use of time.

**Comment by jeffz6** [comment]
Your Name: Jeffrey Zhang

Overview of Lab Comments: I think this lab does a good job of explaining recursion, which is probably one of the harder concepts for new programmers to 
grasp.  The amount of hand-holding at the start seems just right, and gets reduced to about nothing, although I would say that the lack of help in the C-
curve part is too sudden.

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/recursive-tree-part-1.html?topic=berkeley_bjc/recur/recursion-trees-

fractals.topic&step=0&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html
Type of Issue: clarity
Severity: minor
Comments: Why are the tree blocks purple (Look)?  From what I understand, they'd normally be blue (Motion)

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/vary-your-tree.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-

fractals.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=5
Type of Issue: ordering
Severity: minor
Comments: This comes right after a quiz that requires the students to understand what these modifications to the script will do.  Shouldn't this come 
before?  The quiz would be much more doable if students had a chance to play around with the code first.

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/base-cases.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-

fractals.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=7
Type of Issue: ordering
Severity: minor
Comments: It feels weird to completely break down recursion after the students have finished most of the lab.

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/fractals/c-curve.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-

fractals.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=9
Type of Issue: pacing
Severity: moderate
Comments: This is a big jump up from the snowflake in terms of how much the students need to do themselves.  While they do need to learn it, this feels 
like we're dropping them off a cliff.

**Comment by nish2020** [comment]
Your Name: Nishchala Singhal

Overview of Lab Comments: Overall, this is a good lab that clearly explains tree recursion, a concept that can be difficult for students to grasp at first. I like how the lab guides students through the first few problems and then slowly lets them do it themselves. I do think, however, that the anatomy of a recursive function could come a little earlier in the lab, since right now it gets explained pretty late. That way, students may have some more context/background of what they're trying to accomplish. 

For specific issues: 

Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/base-cases.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=7
Type of Issue:  Lack of Clarity
Severity: Moderate
Comments: The part where it describes the anatomy of a recursive function is a little unclear (especially the inner bullet points). I think that students may tend to skip over it/not absorb it that well. Maybe some pictures of the different stages/steps could help? 

Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/recursive-tree-part-1.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=0fractals.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=7
Type of Issue:  Lack of Emphasis
Severity: Minor
Comments: Where it says "We're going to create a tree block in Snap!. It'll start with a move block to draw the trunk of the tree, then a turn block turning left, then a tree block to draw the left smaller tree," I think it should emphasize the use of the 'tree' block a bit more, since that's the main idea of recursion.

**Comment by valenzuelav94** [comment]
Your Name: Vanessa Valenzuela 

Overall of Lab Comments: This is, undoubedtly, by far one of my favorite cs10 labs. Not only does it cover recursion exlicitly well, it also provides visuals and step-by-step instructions on how to go around the fundamental steps that will allow you to understand the basics of the process. 

For Specific Issues: 

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/random-tree.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=6

Type of Issue: Clarity  

Severity: Minor 

Comment(s): As I was working on this lab individually I found myself having a hard time remembering where to start. I noticed that many of the students in the labs had questions on this particular section as well, primarily because it was asking for too much (such as: whether or not the direction of the sprite matters if they have a core understanding of recursion, or if it's even important to be able to randomize a recursive pattern). As for myself, I had a hard time trying to get my sprite to go to the direction it was intended to go to. If anything it needs more direction because if this is the first time students are seeing recursion then they're going to be discouraged with the amount of time it takes to tackle this particular portion of the lab.

**Comment by mfujimoto** [comment]
Your Name: Mason Fujimoto

Overview of Lab Comments: Definitely one of the most challenging labs of the year. Recursion is a difficult and powerful concept. I think one way we can address a student's potential difficulty in understanding this concept is by offering different ways to visualize recursion. Overall I think that this lab will click for half of the students, but needs some extra media to help those who have a hard time wrapping their heads around the subject. Dan likes to go step by step through a recursion code for it to work. I found that practice to be useful when done once, but on the quest and in lab, I think a student should learn other ways to handle code with recursion. I would like to see an emphasis on diagrams or animations explaining which part of the code is doing what in a visual recursion example. However, that might be wishful thinking.

For specific issues:

Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/treepart3.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=2
Type of Issue: pacing
Severity: minor
Comments: The bottom of the page suggests a student try to make a tree like the one pictured (green leaves brown wood and randomness). I think that this is another case of a tedious exercise where the time spent doing it is best spent on trying to understand different cases of recursion rather than focus on the tree example too much.

**Comment by adambrown13** [comment]
Your Name: Adam Brown

Overview of Lab Comments: I think this lab does a great job of helping students through all the topics. It's full of manageable, yet still challenging problems. However, the problem at the very end seems to be a little outside the curve of the rest of the lab. Besides that, a very put together lab

Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/base-cases.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=7
Type of Issue: Clarity
Severity: Moderate
Comments: I think the motions behind how recursion work are sorta glossed over here. Since they're going to be using recursion so much in the rest of the course, it could really benefit from a more step by step delineated explanation

**Comment by carenthomas** [comment]
Your Name: Caren Thomas

Overview of Lab Comments: This lab was the most difficult for me when I took the class and I'm glad to see that it has greatly improved. However, for students struggling with recursion, a little more guidance may be required that that provided. I also feel that there should be more activities that focus on practice with recursion - during lab, lots of students were concerned about minor details with how their trees looked, etc. which wasted a lot of time for them.

Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/base-cases.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=7
Type of Issue: Feature
Severity: Low Priority
Comments: This is the part that trips up most students, so special attention should be given here. The labels that correspond to each line are helpful, but maybe some sort of gif or graphic alongside it to more visually explain the steps would be nice.

Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/recursive-tree-part-2.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=1
Type of Issue: Word choice, lack of clarity
Severity: Minor inconvenience
Comments: The language used in the sentence "So it makes sense to wonder if we can't replace them all with a single tree block" was not very clear.

**Comment by MicBrain** [comment]
Rafayel Mkrtchyan

Overview: This lab was pretty challenging and time-consuming for a lot of students. This was one of the labs where students were asking a lot of questions. The ideas of this lab are pretty hard to understand at the beginning and that is the reason that I think we need to make the material more comprehensive for the next semester. 

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/recur/tree/random-tree.html?topic=berkeley_bjc%2Frecur%2Frecursion-trees-fractals.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=6
Issue: Clarity, hard to understand.
Comments: This was probably the hardest part of the lab for a lot of students. It took me a while to figure out the best way to explain that to them. I think we need to provide them more instructions for approaching this problem from the right side.
